rest-service/manager_rest/cluster_status_manager.py

```diff
@@ -303,6 +304,56 @@ def _get_db_cluster_status(db_service, expected_nodes_number):
     return db_service[STATUS]
 
 
+def _should_not_validate_cluster(service):
+    return service[STATUS] == ServiceStatus.FAIL or service[IS_EXTERNAL]
+
+
+def _get_broker_cluster_status(broker_service, expected_nodes_number):
+    """
+    Get the status of the broker cluster. The majority of nodes should be
+    running and recognizing the same cluster, 'Healthy' cluster means all of
+    them do.
+    """
+    if _should_not_validate_cluster(broker_service):
+        return broker_service[STATUS]
+
+    broker_nodes = broker_service['nodes']
+    active_broker_nodes = {name: node for name, node in broker_nodes.items()
+                           if node[STATUS] == ServiceStatus.HEALTHY}
+
+    if not _verify_broker_cluster_status(active_broker_nodes):
+        return ServiceStatus.FAIL
+
+    if expected_nodes_number != len(active_broker_nodes):
+        return ServiceStatus.DEGRADED
+
+    return broker_service[STATUS]
+
+
+def _verify_broker_cluster_status(active_broker_nodes):
+    """
+    Checks whether the cluster status is the same on all active nodes.
+    If yes, returns True, else, logs which nodes don't recognize the same
+    cluster and returns False.
+    """
+    first_node_name = ''
+    first_node_cluster_status = {}
+
+    for node in active_broker_nodes.values():
+        broker_service = node[SERVICES][BROKER_SERVICE_KEY]
+        cluster_status = broker_service[EXTRA_INFO]['cluster_status']
+        cluster_nodes_status = cluster_status['cluster_nodes_status']
+        node_name = cluster_status['node_name']
+        if not first_node_cluster_status:
+            first_node_cluster_status = cluster_nodes_status
+            first_node_name = node_name
+            continue
+        if cluster_nodes_status != first_node_cluster_status:
+            current_app.logger.info('{0} is not in the same cluster as '
+                                    '{1}'.format(first_node_name, node_name))
+            return False
```

Comment:
let's run over all nodes, and return the failure at the end with the full report on what went wrong